coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/transformers/FromUtcTimestampOperatorTransformer.java

```diff
@@ -122,12 +120,6 @@ protected SqlCall transform(SqlCall sqlCall) {
   }
 
   private SqlCall castOperand(SqlNode operand, RelDataType relDataType) {
-    return SqlStdOperatorTable.CAST.createCall(ZERO, operand, getSqlDataTypeSpecForCasting(relDataType));
-  }
-
-  private SqlDataTypeSpec getSqlDataTypeSpecForCasting(RelDataType relDataType) {
```

Comment:
could you add an example of Coral translation for before & after this code change? Since there's no difference in unit tests, does that mean this is a bug or an unnecessary transformation?

Reply:
> could you add an example of Coral translation for before & after this code change

After adding in the precisions, the first case in `testSubstrWithTimestampOperator` in `HiveToTrinoConverterTest` started to fail because we were creating a faulty type spec that fails [this assertion ](https://jarvis.corp.linkedin.com/codesearch/result/?name=SqlBasicTypeNameSpec.java&path=linkedin-calcite%2Fcore%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fcalcite%2Fsql&reponame=linkedin%2Flinkedin-calcite#219)in the Calcite's derive type function for `SqlBasicTypeNameSpec`. Therefore `getSqlDataTypeSpecForCasting` is a bug, if you look at  [`SqlTypeUtil.convertTypeToSpec`](https://jarvis.corp.linkedin.com/codesearch/result/?name=SqlTypeUtil.java&path=linkedin-calcite%2Fcore%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fcalcite%2Fsql%2Ftype&reponame=linkedin%2Flinkedin-calcite#1023) there are more involved steps that involve using precision/scale values.